### PR TITLE
Add new network template

### PR DESCRIPTION
**Pull request by cjeanner**
With the new CI layout, having a dedicated nodeset for CRC alongside a
dedicated Compute, we need to get a dedicated network template file in
order to consume the right interface - eth1, aka nic2.

That "nic2" interface is associated to a private, dedicated network in
order to allow network isolation to proceed, exactly like late OVB jobs
in TripleO CI ecosystem.

This patch therefore adds a new "ci" dedicated template we can then
consume from within the ci-framework, ensuring we're using the correct
interface.

Note: this is hopefully a temporary workaround - the ansibleVars we
should use is, for now, a string, and we can't edit it with kustomize.
There's a plan to change its type, allowing proper edition of the
mapping.

Signed-off-by: Cédric Jeanneret <cjeanner@redhat.com>

### CODE REVIEW

File: roles/edpm_network_config/templates/ci/extracted_crc_nic2_vlans.j2
Comment (rabi) [inline roles/edpm_network_config/templates/ci/extracted_crc_nic2_vlans.j2:19]: > order to consume the right interface - eth1, aka nic2.

You don't need a separate template in the role added for CI for that. It can be set with edpm_os_net_config_mappings/net_config_data_lookup ansible var[1].

[1] https://github.com/openstack-k8s-operators/edpm-ansible/blob/main/plugins/modules/edpm_os_net_config_mappings.py#L67-L69

**Review by rabi — CHANGES_REQUESTED** [review]

File: roles/edpm_network_config/templates/ci/extracted_crc_nic2_vlans.j2
Comment (cjeanner) [reply roles/edpm_network_config/templates/ci/extracted_crc_nic2_vlans.j2:19]: hm, overriding the "meaning" of nic1 then?

**Comment by cjeanner** [comment]
Following @rabi 's indication, ci-framework now pushes a small snippet adding the mapping. I can close this PR then. Thanks for the hint!

**Comment by cjeanner** [comment]
After some discussions and checks, we can't use kustomize against the `ansibleVars` because it's basically a string:
https://github.com/openstack-k8s-operators/dataplane-operator/blob/main/api/v1beta1/common.go#L54

There's apparently plan to change this parameter type, but until it's done, we have to get our proper mapping in in order to move on and get the new layout working.

Once the parameter type is changed, we'll be able to leverage kustomize in order to make the needed changes.

File: roles/edpm_network_config/templates/ci/extracted_crc_nic2_vlans.j2
Comment (cjeanner) [reply roles/edpm_network_config/templates/ci/extracted_crc_nic2_vlans.j2:19]: As written bellow - we can't use that for now, because of the type associated to `ansibleVars`... basically, we'd override the whole value, because it's a string, not a proper mapping :(.

**Comment by rabi** [comment]
> After some discussions and checks, we can't use kustomize against the ansibleVars because it's basically a string

I think just add a change to install_yamls[1] add the var should be enough

**Review by rebtoor — APPROVED** [review]

**Comment by cjeanner** [comment]
/hold
`error: timed out waiting for the condition on openstackdataplanes/openstack-edpm`

We want to test that patch in ci-framework before landing it - that's done here mostly: https://github.com/openstack-k8s-operators/ci-framework/pull/361

**Comment by cjeanner** [comment]
/unhold

That patch is working as-is - we're facing some "next issue" related to DNS, but it has nothing to do with this new file. So we're good to go.

@rebtoor @rabi if any of you could merge this one, even if zuul is crying (there's no reason it cries on that file addition - issues are the usual ones we want to correct with the new job layout), that would be really nice :).

**Review by rebtoor — APPROVED** [review]

**Pull request merged**
